executor/aggfuncs/func_value_test.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aggfuncs_test
+
+import (
+	"unsafe"
+
+	. "github.com/pingcap/check"
+	"github.com/pingcap/parser/ast"
+	"github.com/pingcap/parser/mysql"
+	"github.com/pingcap/tidb/executor/aggfuncs"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
+)
+
+func getEvaluatedMemDelta(row *chunk.Row, dataType *types.FieldType) (memDelta int64) {
+	memDelta = 0
+	switch dataType.Tp {
+	case mysql.TypeString:
+		memDelta = int64(unsafe.Sizeof(row.GetString(0)))
```

Comment:
ditto, sizeof(string) always is 16. And could we create some unit tests with no constant memDetla?

Reply:
Thanks a lot! I've changed the measurement both for json and string from ```sizeof``` to ```len```.
But as for "non-zero memDelta tests", I did not quite get it. Since the ```genSrcChunk``` method will generate ```"%d"``` for each string example, whose length will be 1 in the unit tests at presence. Could you please explain it a little bit further for me? Thanks very much!

Reply:
That is my mistake, the memDetla of the value aggFunc is the size of the first row. So I think we don't need to add more unit tests for value aggFunc anymore.